### ISSUE TITLE
Some question about lr and warm up

### ISSUE THREAD

**Issue by WangLanxiao** [YehLi/xmodaler#15]
Hello, after I accidentally interrupted the training, I used --resume to resume the training, but I found that the learning rate has not been restored. How can I correct such an error.
In addition, which parameters control the end time of warm up?
Thanks a lot！
![image](https://user-images.githubusercontent.com/34602362/131781412-a45ebe8d-c8a5-40f9-8ca4-af4088a29f23.png)

**Comment by WangLanxiao** [comment]
The red line represents the learning rate after the training is interrupted

**Comment by jianjieluo** [comment]
hi @WangLanxiao ,

1. Since we don't dump lr_scheduler's state in the checkpoint file by default, you can not restore the learning rate schedule after training interruption. I have updated our code to support dump&load lr_scheduler in the checkpoint file, and you can refer to [this commit](https://github.com/YehLi/xmodaler/commit/5aa9de5c0c8bd1404796074ebbfb46de8500ebf5) for more information.

2. You can use `LR_SCHEDULER.WARMUP` to control the end time of warm-up. The number of  `LR_SCHEDULER.WARMUP` indicates the warmup **epoch number** by default except `NoamLR` which represents the warmup **iteration number**. You can also refer to the implementation of [noam_lr](https://github.com/YehLi/xmodaler/blob/5aa9de5c0c8bd1404796074ebbfb46de8500ebf5/xmodaler/lr_scheduler/noam_lr.py#L29) and [other warmup schedulers](https://github.com/YehLi/xmodaler/blob/5aa9de5c0c8bd1404796074ebbfb46de8500ebf5/xmodaler/lr_scheduler/warmup_lr.py#L30) for more information.

Thanks for your active usage of Xmodaler to make it better!

Best,
Jianjie

**Comment by WangLanxiao** [comment]
Thank you very much for your reply~

**Comment by jianjieluo** [comment]
Closed due to inactivity.
